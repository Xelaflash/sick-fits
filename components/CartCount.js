import { CSSTransition, TransitionGroup } from 'react-transition-group';
import styled from 'styled-components';

const Dot = styled.div`
  background: var(--red);
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;
  /* below makes all numbers same width */
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;
`;

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: transform 0.6s;
    backface-visibility: hidden;
  }
  .count-enter {
    transform: scale(3.2) rotateY(0.5turn);
  }
  .count-enter-active {
    transform: rotateY(0);
  }
  .count-exit {
    position: absolute;
    top: 0;
    transform: rotateY(0);
  }
  .count-exit-active {
    transform: scale(3.2) rotateY(0.5turn);
  }
`;

export default function CartCount({ count }) {
  return (
    <AnimationStyles>
      <TransitionGroup>
        <CSSTransition
          unmountOnExit
          className="count"
          classNames="count"
          key={count}
          timeout={{ enter: 600, exit: 600 }}
        >
          <Dot>{count}</Dot>
        </CSSTransition>
      </TransitionGroup>
    </AnimationStyles>
  );
}
