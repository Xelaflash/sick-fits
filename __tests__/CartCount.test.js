import { render, waitFor } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('1. Renders', () => {
    render(<CartCount count={10} />);
  });
  it('2. matches snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it('3. updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11');
    // exactly the same
    // expect(container).toHaveTextContent('11');
    // ! Update the props
    rerender(<CartCount count={12} />);
    // we need to wait for cart count animation to finish, because for a short moment it will have old & new value causing test to fail
    // await wait(600);
    await waitFor(() => {
      expect(container.textContent).toBe('12');
    });
    expect(container).toMatchSnapshot();
    // debug();
  });
});
