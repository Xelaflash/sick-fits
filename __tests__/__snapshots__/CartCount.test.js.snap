// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> 2. matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 bNVMoB"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> 3. updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 bNVMoB"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 iephzA count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
